neuro_san/internals/run_context/langchain/core/langchain_run_context.py

```diff
@@ -613,15 +602,14 @@ async def wait_on_run(self, run: Run, journal: Journal = None) -> Run:
         # Attempt to count tokens/costs while invoking the agent.
         llm: BaseLanguageModel = self.llm_resources.get_model()
         token_counter = LangChainTokenCounter(llm, self.invocation_context, self.journal)
-        await token_counter.count_tokens(self.ainvoke(agent_executor, inputs, invoke_config))
+        await token_counter.count_tokens(self.ainvoke(inputs, invoke_config), max_execution_seconds)
```

Comment:
We now pass max_execution_seconds as an arg to the token counter.
In there a task is created, and we use this value as a timeout when we wait for the task to complete.